FEM/Output.cpp

```diff
@@ -1467,6 +1530,51 @@ void COutput::ELEWriteDOMDataTEC()
     tec_file.close();  // kg44 close file
 }
 
+void COutput::WriteELECellCenteredValuesTECHeader(fstream& tec_file,
+                                      int e_type, string ele_type_name)
+{
+        // OK411
+    size_t no_elements = 0;
+
+    const size_t mesh_ele_vector_size(m_msh->ele_vector.size());
+    for (size_t i = 0; i < mesh_ele_vector_size; i++)
+        if (m_msh->ele_vector[i]->GetMark())
+            if (m_msh->ele_vector[i]->GetElementType() == e_type)
+                no_elements++;
+    //--------------------------------------------------------------------
+    // Write Header I: variables
+    tec_file << "VARIABLES = \"X\",\"Y\",\"Z\",\"VX\",\"VY\",\"VZ\"";
```

Comment:
is it ok to assume there is always output of the velocity? For example, mechanics simulations will not have the values.

Reply:
Good point. Honestly, I do not know. I added an adapted copy of the existing element output and did not look further into the existing logic. I expected that anything that worked before would be available in the new format then.

However, I should test it with a pure mechanics simulation and see if a problem appears. Could you/anybody point out which mechanics quantities are available for element output or maybe where this information can be opbtained (in the code)?

Reply:
i see. if you have just copied the existing code, you don’t need to test it.